chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -130,6 +193,96 @@ void BraveRenderViewContextMenu::AddSpellCheckServiceItem(
   // Suppress adding "Spellcheck->Ask Brave for suggestions" item.
 }
 
+bool BraveRenderViewContextMenu::IsIPFSCommandIdEnabled(int command) const {
+  if (!IsIpfsMenuEnabled(browser_context_))
+    return false;
+  switch (command) {
+    case IDC_CONTENT_CONTEXT_IMPORT_IPFS:
+      return true;
+    case IDC_CONTENT_CONTEXT_IMPORT_IPFS_PAGE:
+      return source_web_contents_->GetURL().SchemeIsHTTPOrHTTPS();
+    case IDC_CONTENT_CONTEXT_IMPORT_IMAGE_IPFS:
+      return params_.has_image_contents;
+    case IDC_CONTENT_CONTEXT_IMPORT_VIDEO_IPFS:
+      return content_type_->SupportsGroup(
+          ContextMenuContentType::ITEM_GROUP_MEDIA_VIDEO);
+    case IDC_CONTENT_CONTEXT_IMPORT_AUDIO_IPFS:
+      return content_type_->SupportsGroup(
+          ContextMenuContentType::ITEM_GROUP_MEDIA_AUDIO);
+    case IDC_CONTENT_CONTEXT_IMPORT_LINK_IPFS:
+      return !params_.link_url.is_empty();
+    case IDC_CONTENT_CONTEXT_IMPORT_SELECTED_TEXT_IPFS:
+      return !params_.selection_text.empty() &&
+             params_.media_type == ContextMenuDataMediaType::kNone;
+    default:
+      NOTREACHED();
+  }
+  return false;
+}
+
+#if BUILDFLAG(IPFS_ENABLED)
+void BraveRenderViewContextMenu::SeIpfsIconAt(int index) {
+  auto& bundle = ui::ResourceBundle::GetSharedInstance();
+  const auto& ipfs_logo = *bundle.GetImageSkiaNamed(IDR_BRAVE_IPFS_LOGO);
+  ui::ImageModel model = ui::ImageModel::FromImageSkia(ipfs_logo);
+  menu_model_.SetIcon(index, model);
+}
+
+void BraveRenderViewContextMenu::BuildIPFSMenu() {
+  if (IsIpfsMenuEnabled(browser_context_)) {
```

Comment:
Nit: might be visually easier to read if `!`IsIpfsMenuEnabled is checked and return right away instead of wrapping the entire method content into an `if`.